client/signup/main.jsx

```diff
@@ -185,13 +191,13 @@ const Signup = React.createClass( {
 	componentDidMount() {
 		debug( 'Signup component mounted' );
 		SignupProgressStore.on( 'change', this.loadProgressFromStore );
-		SignupProgressStore.on( 'change', this.loadDependenciesFromStore );
+		this.unsubscribeSignupDepStoreListener = SignupDependencyStore.reduxStore.subscribe( this.loadDependenciesFromStore );
```

Comment:
I originally tried to check if the subtree has been updated before updating state in 66e1ee3ab5c294f2de64a0e2329e548885504c5a and it seemed to be working as expected - re-renders happen only if the subtree has been updated.

But I wanted to check out if there was a better way to handle this and I think I managed to improve my React-Redux skills a bit in 2fc8a25a8c087c0c42b825389991d461e74d588c. 

The component wrapper created by `connect` takes care of exactly what we want to achieve here - it only updates the property that is passed to the component if an actual update has happened to the subtree.  This requires the data to be passed as a `prop` to the component, but the end result seems better, since we don't have to handle all the update logic ourselves. 

The end result for both is the same - re-renders happen only when there is update to the subtree and not the whole global state.